### PR TITLE
Add support for PHP 8

### PR DESCRIPTION
**Pull request by lepiaf**

### CODE REVIEW

**Comment by lepiaf** [comment]
PR is blocked by dependency https://github.com/php-http/curl-client

```
 composer update --prefer-dist --no-progress --no-suggest $COMPOSER_FLAGS --ansi

You are using the deprecated option "--no-suggest". It has no effect and will break in Composer 3.

Loading composer repositories with package information

Updating dependencies

Your requirements could not be resolved to an installable set of packages.

  Problem 1

    - php-http/curl-client[2.0.0, ..., 2.1.1] require php ^7.1 -> your php version (8.0.0) does not satisfy that requirement.

    - Root composer.json requires php-http/curl-client ^2.0 -> satisfiable by php-http/curl-client[2.0.0, 2.1.0, 2.1.1].
```

There is already a PR to support PHP 8. We have to wait a new release of this package.
- https://github.com/php-http/curl-client/pull/70
- https://github.com/php-http/curl-client/issues/68

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
         }
     ],
     "require": {
-        "php": "^7.2",
+        "php": "^7.2 || ^8.0",
```
Comment (odolbeau) [inline composer.json:14]: Could you use `>7.2.5` as in symfony please?

**Comment by odolbeau** [comment]
Thanks for your contribution! :ok_hand: 
Let's wait for curl then. :)

**Comment by lepiaf** [comment]
this PR https://github.com/php-http/curl-client/pull/71 superseded https://github.com/php-http/curl-client/pull/70

**Comment by lepiaf** [comment]
https://github.com/php-http/curl-client/pull/71 has been merged, A new version has been released https://github.com/php-http/curl-client/releases/tag/2.2.0

**Comment by odolbeau** [comment]
Perfect, let's merge! :ok_hand:

**Pull request merged**